packages/core/src/interpreter.ts

```diff
@@ -973,7 +972,7 @@ export class Interpreter<
 
     childService
       .onDone((doneEvent) => {
-        this.removeChild(childService.id);
+        this.stopChild(childService.id);
```

Comment:
in a sense, this will now send a "stop" event to the already stopped child, which doesn't sound ideal, even if a stopped child should just ignore all newly received events

Reply:
If a child is removed, it should be stopped, though.

Reply:
If we remove a child without sending a "stop" signal, the child will be orphaned, which is a memory leak.

Reply:
this is true for other scenarios - yes, but in here we receive a signal from the child that it has stopped, so there is no need to send a "stop" signal to it, the child is already stopped - it has just notified us about this